### ISSUE TITLE
[Chore] Unable to run app using Codespaces Instructions

### ISSUE THREAD

**Issue by DanWahlin** [Azure-Samples/contoso-real-estate#449]
**Are you accessing the project from the GitHub Codespaces?**

- [ ] No, I am running the project on my local machine (I will fill in the OS section below)
- [X] Yes, I am using GitHub Codespaces.

**Describe the bug**

Not all of the servers start. I'm seeing this in the logs after running npm install && npm start in the codespace:

```
[services] postgres_setup-1  | PostgreSQL Database restored successfully
[services] postgres_setup-1 exited with code 0
[services] blog-1            | TypeError: fetch failed
[services] blog-1            |     at Object.fetch (node:internal/deps/undici/undici:11731:11)
[services] blog-1            |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
[services] blog-1            |   cause: Error: getaddrinfo ENOTFOUND blog-cms
[services] blog-1            |       at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:107:26)
[services] blog-1            |       at GetAddrInfoReqWrap.callbackTrampoline (node:internal/async_hooks:128:17) {
[services] blog-1            |     errno: -3008,
[services] blog-1            |     code: 'ENOTFOUND',
[services] blog-1            |     syscall: 'getaddrinfo',
[services] blog-1            |     hostname: 'blog-cms'
[services] blog-1            |   }
[services] blog-1            | }
[services] blog-1            | [Error: An error occurred in the Server Components render. The specific message is omitted in production builds to avoid leaking sensitive details. A digest property is included on this error instance which may provide additional details about the nature of the error.] {
[services] blog-1            |   digest: '2178396693'
[services] blog-1            | }
[services] blog-1            | TypeError: fetch failed
[services] blog-1            |     at Object.fetch (node:internal/deps/undici/undici:11731:11)
[services] blog-1            |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
[services] blog-1            |   cause: Error: getaddrinfo ENOTFOUND blog-cms
[services] blog-1            |       at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:107:26)
[services] blog-1            |       at GetAddrInfoReqWrap.callbackTrampoline (node:internal/async_hooks:128:17) {
[services] blog-1            |     errno: -3008,
[services] blog-1            |     code: 'ENOTFOUND',
[services] blog-1            |     syscall: 'getaddrinfo',
[services] blog-1            |     hostname: 'blog-cms'
[services] blog-1            |   }
[services] blog-1            | }
[services] blog-1            | [Error: An error occurred in the Server Components render. The specific message is omitted in production builds to avoid leaking sensitive details. A digest property is included on this error instance which may provide additional details about the nature of the error.] {
[services] blog-1            |   digest: '2178396693'
[services] blog-1            | }
[services] blog-1            | TypeError: fetch failed
[services] blog-1            |     at Object.fetch (node:internal/deps/undici/undici:11731:11)
[services] blog-1            |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
[services] blog-1            |   cause: Error: getaddrinfo ENOTFOUND blog-cms
[services] blog-1            |       at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:107:26)
[services] blog-1            |       at GetAddrInfoReqWrap.callbackTrampoline (node:internal/async_hooks:128:17) {
[services] blog-1            |     errno: -3008,
[services] blog-1            |     code: 'ENOTFOUND',
[services] blog-1            |     syscall: 'getaddrinfo',
[services] blog-1            |     hostname: 'blog-cms'
[services] blog-1            |   }
[services] blog-1            | }
[services] blog-1            | [Error: An error occurred in the Server Components render. The specific message is omitted in production builds to avoid leaking sensitive details. A digest property is included on this error instance which may provide additional details about the nature of the error.] {
[services] blog-1            |   digest: '2178396693'
[services] blog-1            | }
```

**To Reproduce**

1. Go to Codespace
2. Type in command npm install && npm start
3. View log errors

**Expected behavior**
Expected all services to start

**OS (please complete the following information):**

- OS: MacOS
- OS Version: Sonoma 
- npm version: 9.5.0 (codespaces version)
- Node.js version: 18.15.0 (codespaces version)
- output of `azd version`: 1.6.1 (commit eba2c978b5443fdb002c95add4011d9e63c2e76f) - (codespaces version)

**Comment by juliamuiruri4** [comment]
Facing same issue, though this just affects the blog cms service

**Comment by ZedFarNl** [comment]
encounter the same issue in codespace

**Comment by DanWahlin** [comment]
Thanks for the information @ZedFarNl and sorry you’re hitting an issue here. Are you seeing the exact same error message as above or something different? If you can share the specific error that would be helpful.

//cc @manekinekko and @anfibiacreativa

**Comment by anfibiacreativa** [comment]
Thank you for reporting! We will be prioritizing this issue.

**Comment by manekinekko** [comment]
I've traced this issue and found the problem. It seems to be related to https://github.com/strapi/strapi/issues/12493 that prevents the CMS from starting, which explain why we are getting a 503. Since the Blog depends on the CMS, it's also impacted, hence the `Error: getaddrinfo ENOTFOUND blog-cms` error.

In the meantime, in order to unblock everyone, I've disabled [this](https://github.com/Azure-Samples/contoso-real-estate/blob/cba7190a068caa4249ef565ef3b6db50ee5a1c7f/docker-compose.yml#L54) configuration in the Docker compose file untill the Strapi issue is fixed.

![image](https://github.com/Azure-Samples/contoso-real-estate/assets/1699357/1c0a8eb3-6b6f-401b-a876-8c87facc5f7d)
![image](https://github.com/Azure-Samples/contoso-real-estate/assets/1699357/43e243c5-78a2-4c1c-89d6-39e7513aaa2b)
![image](https://github.com/Azure-Samples/contoso-real-estate/assets/1699357/982c223d-acbd-474d-a8b3-69ccf554ff9f)


Note: the broken images in Blog are an expected result for now, because the db restore script contains image links that were hosted on Azure when the original db dump has been made. This can be improved in a separate PR if someone wants to take a shot and clean the db script.

**Comment by manekinekko** [comment]
Updates: I kept looking at this issue, and it seems that the CMS server fails to start randomly, I assuming this is because of some race condition issues when starting all containers at once. To fix this for now, please restart the CMS container and the server should start:

```
docker compose restart blog-cms
```
